packages/kit/src/exports/vite/index.js

```diff
@@ -154,6 +154,15 @@ export async function sveltekit() {
 		...svelte_config.vitePlugin
 	};
 
+	// TODO: remove this after some migration period. perhaps mention this migration in the SvelteKit
+	// 3 migration guide and remove the warning then
+	const pkg = JSON.parse(fs.readFileSync(path.join(cwd, 'package.json'), 'utf-8'));
+	if (!pkg.scripts?.prepare?.includes('svelte-kit')) {
+		svelte_config.onwarn(
```

Comment:
calling onwarn yourself would have to be guarded with `svelte_config?.onwarn` but its not a given that a user has configured it, or does sveltekit provide one always?

in rollup (and i assume this also works in vite) you'd call `this.warn` inside hooks to send a warning through onwarn

https://rollupjs.org/plugin-development/#this-warn

(or its more modern sibling onLog)

Reply:
thanks. updated it to use `this.warn`